src/spec.ts

```diff
@@ -591,24 +590,26 @@ function normalizePathOverlay(spec: NormalizedUnitSpec, config: Config = {}): No
   }
 
   if (lineOverlay) {
-    const {interpolate} = markDef;
+    const lineMarkDef = omit(markDef, ['line']);
```

Comment:
I don't think we should really inherit everything from the main mark.  For example, if you set the main area mark's opacity, it should not affect the overlay line's opacity.  Similar if you set the main line mark's opacity, it should not affect overlay point's opacity.  

Plus, separate them mean you can customize both of them. If you make the overlay derive all properties, then we require user to specify properties of both the main layer and the overlay layer in order to set *only* the main layer.  I think that's an undesired behavior.

Instead you could have just inherited `clip`.

Reply:
Okay, I will inherit clip and interpolate.